ksql-engine/src/main/java/io/confluent/ksql/function/InternalFunctionRegistry.java

```diff
@@ -233,21 +233,39 @@ private void addMathFunctions() {
           AbsKudf.class));
 
       addBuiltInFunction(KsqlFunction.createLegacyBuiltIn(
-          Schema.OPTIONAL_FLOAT64_SCHEMA,
+          Schema.OPTIONAL_INT64_SCHEMA,
           Collections.singletonList(Schema.OPTIONAL_FLOAT64_SCHEMA),
-          "CEIL",
+          CeilKudf.NAME,
           CeilKudf.class));
 
       addBuiltInFunction(KsqlFunction.createLegacyBuiltIn(
           Schema.OPTIONAL_FLOAT64_SCHEMA,
+          ImmutableList.of(Schema.FLOAT64_SCHEMA, Schema.OPTIONAL_INT32_SCHEMA),
+          CeilKudf.NAME,
+          CeilKudf.class));
+
+      addBuiltInFunction(KsqlFunction.createLegacyBuiltIn(
+          Schema.OPTIONAL_INT64_SCHEMA,
           Collections.singletonList(Schema.OPTIONAL_FLOAT64_SCHEMA),
-          "FLOOR",
+          FloorKudf.NAME,
+          FloorKudf.class));
+
+      addBuiltInFunction(KsqlFunction.createLegacyBuiltIn(
+          Schema.OPTIONAL_FLOAT64_SCHEMA,
+          ImmutableList.of(Schema.FLOAT64_SCHEMA, Schema.OPTIONAL_INT32_SCHEMA),
+          FloorKudf.NAME,
           FloorKudf.class));
 
       addBuiltInFunction(KsqlFunction.createLegacyBuiltIn(
           Schema.OPTIONAL_INT64_SCHEMA,
           Collections.singletonList(Schema.OPTIONAL_FLOAT64_SCHEMA),
-          "ROUND",
+          RoundKudf.NAME,
+          RoundKudf.class));
+
+      addBuiltInFunction(KsqlFunction.createLegacyBuiltIn(
+          Schema.OPTIONAL_FLOAT64_SCHEMA,
+          ImmutableList.of(Schema.FLOAT64_SCHEMA, Schema.OPTIONAL_INT32_SCHEMA),
```

Comment:
likewise,

```suggestion
          ImmutableList.of(Schema.OPTIONAL_FLOAT64_SCHEMA, Schema.OPTIONAL_INT32_SCHEMA),
```

Reply:
As mentioned before...

```suggestion
          ImmutableList.of(Schema.OPTIONAL_FLOAT64_SCHEMA, Schema.OPTIONAL_INT32_SCHEMA),
```